bgpd/bgp_nexthop.c

```diff
@@ -816,94 +890,157 @@ static void bgp_show_nexthops_detail(struct vty *vty, struct bgp *bgp,
+			vty_out(vty, "\n");
+		}
+		bgp_show_nexthops_detail(vty, bgp, bnc, json);
 	} else if (CHECK_FLAG(bnc->flags, BGP_NEXTHOP_EVPN_INCOMPLETE)) {
-		vty_out(vty,
-			" %s overlay index unresolved [IGP metric %d], #paths %d",
-			inet_ntop(bnc->prefix.family, &bnc->prefix.u.prefix,
-				  buf, sizeof(buf)),
-			bnc->metric, bnc->path_count);
-		if (bnc->is_evpn_gwip_nexthop)
-			vty_out(vty, " EVPN Gateway IP");
-		vty_out(vty, "\n");
-		bgp_show_nexthops_detail(vty, bgp, bnc);
+		if (json) {
+			json_object_string_add(json, "ip", buf);
+			json_object_boolean_true_add(json, "valid");
+			json_object_boolean_false_add(json, "complete");
+			json_object_int_add(json, "igpMetric", bnc->metric);
+			json_object_int_add(json, "pathCount", bnc->path_count);
+			if (bnc->is_evpn_gwip_nexthop)
+				json_object_boolean_true_add(json,
+							     "isEvpnGatewayIp");
+		} else {
+			vty_out(vty,
+				" %s overlay index unresolved [IGP metric %d], #paths %d",
+				buf, bnc->metric, bnc->path_count);
+			if (bnc->is_evpn_gwip_nexthop)
+				vty_out(vty, " EVPN Gateway IP");
+			vty_out(vty, "\n");
+		}
+		bgp_show_nexthops_detail(vty, bgp, bnc, json);
 	} else {
-		vty_out(vty, " %s invalid, #paths %d",
-			inet_ntop(bnc->prefix.family, &bnc->prefix.u.prefix,
-				  buf, sizeof(buf)),
-			bnc->path_count);
-		if (peer)
-			vty_out(vty, ", peer %s", peer->host);
-		if (bnc->is_evpn_gwip_nexthop)
-			vty_out(vty, " EVPN Gateway IP");
-		vty_out(vty, "\n");
-		if (CHECK_FLAG(bnc->flags, BGP_NEXTHOP_CONNECTED))
-			vty_out(vty, "  Must be Connected\n");
-		if (!CHECK_FLAG(bnc->flags, BGP_NEXTHOP_REGISTERED))
-			vty_out(vty, "  Is not Registered\n");
+		if (json) {
+			json_object_string_add(json, "ip", buf);
+			json_object_boolean_false_add(json, "valid");
+			json_object_boolean_false_add(json, "complete");
+			json_object_int_add(json, "pathCount", bnc->path_count);
+			if (peer)
+				json_object_string_add(json, "hostname",
+						       peer->host);
+			if (bnc->is_evpn_gwip_nexthop)
+				json_object_boolean_true_add(json,
+							     "isEvpnGatewayIp");
+			if (CHECK_FLAG(bnc->flags, BGP_NEXTHOP_CONNECTED))
+				json_object_boolean_false_add(json,
+							      "isConnected");
+			if (!CHECK_FLAG(bnc->flags, BGP_NEXTHOP_REGISTERED))
+				json_object_boolean_false_add(json,
+							      "isRegistered");
+		} else {
+			vty_out(vty, " %s invalid, #paths %d", buf,
+				bnc->path_count);
+			if (peer)
+				vty_out(vty, ", peer %s", peer->host);
+			if (bnc->is_evpn_gwip_nexthop)
+				vty_out(vty, " EVPN Gateway IP");
+			vty_out(vty, "\n");
+			if (CHECK_FLAG(bnc->flags, BGP_NEXTHOP_CONNECTED))
+				vty_out(vty, "  Must be Connected\n");
+			if (!CHECK_FLAG(bnc->flags, BGP_NEXTHOP_REGISTERED))
+				vty_out(vty, "  Is not Registered\n");
+		}
 	}
 	tbuf = time(NULL) - (monotime(NULL) - bnc->last_update);
-	vty_out(vty, "  Last update: %s", ctime(&tbuf));
+	if (json)
+		json_object_int_add(json, "lastUpdate", tbuf);
+	else
```

Comment:
Would it be possible to print the "lastUpdate" in readable format? May be use json_object_string_add(json, "lastUpdate", ctime(&tbuf))?

Reply:
I am unsure if a  formatter exists for this as well.

Reply:
Fixed